x/liquidstake/abci.go

```diff
@@ -0,0 +1,21 @@
+package liquidstake
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/persistenceOne/pstake-native/v2/x/liquidstake/keeper"
+	"github.com/persistenceOne/pstake-native/v2/x/liquidstake/types"
+)
+
+// BeginBlocker updates liquid validator set changes for the current block
+func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	k.UpdateLiquidValidatorSet(ctx)
+
+	whitelistedValsMap := types.GetWhitelistedValsMap(k.GetParams(ctx).WhitelistedValidators)
+	k.AutocompoundStakingRewards(ctx, whitelistedValsMap)
```

Comment:
Even though there is not IBC involved and looks like a fairly lightweight process, I would not rebalance / autocompound on each block, seems like an overkill to me.

Reply:
It's skipping internally, same as for the rebalancing.

See this reward trigger threshold:
https://github.com/persistenceOne/pstake-native/pull/684/files/2fff5b8fd651bc02fe434ef6f33535f07b218546#diff-2578352b57ab768f4092f755cc00d74602cbe5857540b7b8dfcc4ea39765f164R216-R219